osu.Game/Database/RealmArchiveModelImporter.cs

```diff
@@ -217,7 +217,9 @@ public async Task<ExternalEditOperation<TModel>> BeginExternalEditing(TModel mod
                 // to prevent this bricking external edit, strip invalid characters on external edit.
                 // the presumption here is that whatever produced the mangled archive is primarily at fault here, and we're just trying to trudge on locally as best as possible.
                 // if there are further troubles related to similar issues, reevaluate moving this sort of check to the import side instead (sanitising filenames on import from archive).
-                string destinationPath = Path.Join(mountedPath, realmFile.Filename.GetValidFilename());
+                string destinationPath = mountedPath;
+                foreach (string piece in realmFile.Filename.Split('/').Select(f => f.GetValidFilename()))
```

Comment:
Is this guaranteed to be Unix path? In other words, is this correct to _not_ be `Path.GetFilenameSeparator()` or whatever?